eng/common/scripts/get-codeowners.ps1

```diff
@@ -1,49 +1,50 @@
 param (
-  $TargetDirectory, # should be in relative form from root of repo. EG: sdk/servicebus
-  $RootDirectory, # ideally $(Build.SourcesDirectory)
-  $VsoVariable = "" # target devops output variable
+  [string]$CodeOwnerPathExpression, # Code path to code owners. e.g sdk/core/azure-amqp
+  [string]$ToolVersion = "", # Placeholder. Will update in next PR
+  [string]$ToolPath = "$env:AGENT_TOOLSDIRECTORY", # The place to check the tool existence. Put $(Agent.ToolsDirectory) as default
+  [string]$DevOpsFeed = "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-net/nuget/v3/index.json", # DevOp tool feeds.
+  [string]$CodeOwnerFileLocation = "../../../.github/CODEOWNERS", # The place we fetch CODEOWNERS file
+  [string]$VsoVariable = "" # Option of write code owners into devop variable
 )
-$target = $TargetDirectory.ToLower().Trim("/")
-$codeOwnersLocation = Join-Path $RootDirectory -ChildPath ".github/CODEOWNERS"
-$ownedFolders = @{}
 
-if (!(Test-Path $codeOwnersLocation)) {
-  Write-Host "Unable to find CODEOWNERS file in target directory $RootDirectory"
-  exit 1
+# Check if the retrieve-codeowners tool exsit or not.
+function InstallRetrieveCodeOwnersTool() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    return
+  }
+  Write-Warning "Installing the retrieve-code-owner tool under $ToolPath... "
+  dotnet tool install --tool-path $toolPath --add-source "$DevOpsFeed" --version "$ToolVersion" "Azure.Sdk.Tools.RetrieveCodeOwners"
 }
 
-$codeOwnersContent = Get-Content $codeOwnersLocation
-
-foreach ($contentLine in $codeOwnersContent) {
-  if (-not $contentLine.StartsWith("#") -and $contentLine){
-    $splitLine = $contentLine -split "\s+"
-    
-    # CODEOWNERS file can also have labels present after the owner aliases
-    # gh aliases start with @ in codeowners. don't pass on to API calls
-    $ownedFolders[$splitLine[0].ToLower().Trim("/")] = ($splitLine[1..$($splitLine.Length)] `
-      | ? { $_.StartsWith("@") } `
-      | % { return $_.substring(1) }) -join ","
+# Get the json property from tool command.
+function GetCodeOwnersEntryFromCommand() {
+  if (Get-Command "$ToolPath/retrieve-code-owners") {
+    Write-Error "The retieve-code-owners tool is not properly installed. Please check your tool path. $ToolPath"
   }
+  return & "$ToolPath/retrieve-code-owners" --target-directory "$CodeOwnerPathExpression" --code-owner-file-path "$CodeOwnerFileLocation" 
```

Comment:
Does this even work? I ask because I don't see any processing that would handle the `--target-directory`, I would expect you would need to pass the parameters via position instead of by name.

Reply:
<img width="221" alt="image" src="https://user-images.githubusercontent.com/48036328/142468005-060f9497-4c61-4805-b6b6-7e5f1ad4e88c.png">

We have the parameter and I tested locally

Reply:
Seems like the `Dragonfruit` dependency is what's providing the magic conversion from the function argument names to the CLI flag syntax?

Reply:
I think this is pretty cool but it might be good to put a code comment above the `main` method describing that the arguments will be magically handled by the Dragonfruit dependency.

Reply:
Oh this is interesting and I agree we probably should add a comment there isn't anything to really stop us from removing this dependency and breaking this command line arg parsing.